### PR TITLE
Trim volume size as needed to fit in pool free space

### PR DESCRIPTION
**Pull request by dwlehman**
I've included the test from #189 since it is intended to verify this change.

I am open to a discussion of whether we should expose `MAX_TRIM_PERCENT` in the role defaults. I'm also open to discussion of what the best default value is. I would definitely be open to reducing it to 2-4% as that's more in line with my intended application of this.

### CODE REVIEW

**Comment by dwlehman** [comment]
This should resolve #13

**Comment by dwlehman** [comment]
[citest bad]

**Comment by richm** [comment]
[citest bad]
staging now has the fix so that tests from other PRs won't be used

**Comment by richm** [comment]
[citest bad]

**Comment by nhosoi** [comment]
[citest bad]
(let me rerun the CI to test https://issues.redhat.com/browse/RHELPLAN-67929.)

**Comment by nhosoi** [comment]
[citest pending]

**Comment by nhosoi** [comment]
[citest]

**Comment by nhosoi** [comment]
> [citest bad]
> staging now has the fix so that tests from other PRs won't be used

We could verify that there is no more extra files from previous PRs, e.g., with  `rhel-x/ansible-2.9 (staging) — linux-system-roles-staging-141-jbrp5@2021-02-19 07:06:56.290667`. It ran after the new test-harness was deployed as well as after a pr with extra test (#181).

File: tests/tests_lvm_auto_size_cap.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    - include_role:
+        name: linux-system-roles.storage
+
+    - include_tasks: get_unused_disk.yml
+      vars:
+        min_size: 10g
+        max_return: 1
+
+    - command: lsblk -b -l --noheadings -o NAME,SIZE
+      register: storage_test_lsblk
```
Comment (japokorn) [inline tests/tests_lvm_auto_size_cap.yml:15]: Will adding `changed_when: false` resolve this?

**Review by japokorn — COMMENTED** [review]
One comment, otherwise it looks good to me.

File: tests/tests_lvm_auto_size_cap.yml (hunk 1 above)
Comment (richm) [reply tests/tests_lvm_auto_size_cap.yml:15]: yes - please add `changed_when: false`

**Comment by dwlehman** [comment]
Latest revision added `changed_when: false` to two plays(?) in the tests and a little blurb to the README about the size trimming.

**Pull request merged**